scanner/src/main/kotlin/storages/PostgresStorage.kt

```diff
@@ -193,6 +201,61 @@ class PostgresStorage(
         }
     }
 
+    override fun readInternal(
+        packages: List<Package>,
+        scannerCriteria: ScannerCriteria
+    ): Result<Map<Identifier, List<ScanResult>>> {
+        if (packages.isEmpty()) return Success(emptyMap())
+
+        val minVersionArray = with(scannerCriteria.minVersion) { intArrayOf(major, minor, patch) }
+        val maxVersionArray = with(scannerCriteria.maxVersion) { intArrayOf(major, minor, patch) }
+
+        @Suppress("TooGenericExceptionCaught")
+        return try {
+            val scanResults = runBlocking(Dispatchers.IO) {
+                packages.chunked(max(packages.size / 50, 1)).map { chunk ->
+                    suspendedTransactionAsync {
+                        @Suppress("MaxLineLength")
+                        ScanResultDao.find {
+                            (ScanResults.identifier inList chunk.map { it.id.toCoordinates() }) and
+                                    (rawParam("scan_result->'scanner'->>'name'") tilde scannerCriteria.regScannerName) and
+                                    (rawParam(VERSION_EXPRESSION) greaterEq arrayParam(minVersionArray)) and
+                                    (rawParam(VERSION_EXPRESSION) less arrayParam(maxVersionArray))
+                        }.map { it.identifier to it.scanResult }
+                    }
+                }.flatMap { it.await() }
+                    .groupBy { it.first }
+                    .mapValues { (_, results) -> results.map { it.second } }
+                    .mapValues { (id, results) ->
+                        val pkg = packages.single { it.id == id }
+
+                        results
```

Comment:
Remark: Ideally, this post-processing and filtering of query results would also be done in parallel - but I have no idea whether this is easily possible with the Exposed API.

Reply:
As mentioned in my other comment, this code will be rewritten completely next week, I will keep that in mind and check if it can be run in parallel.